x-pack/plugins/security_solution/cypress/screens/alerts_detection_rules.ts

```diff
@@ -104,7 +112,48 @@ export const INPUT_FILE = 'input[type=file]';
 
 export const TOASTER = '[data-test-subj="euiToastHeader"]';
 
+export const TOASTER_BODY = '[data-test-subj="globalToastList"] .euiToastBody';
+
 export const RULE_IMPORT_OVERWRITE_CHECKBOX = '[id="import-data-modal-checkbox-label"]';
 
 export const RULE_IMPORT_OVERWRITE_EXCEPTIONS_CHECKBOX =
   '[id="import-data-modal-exceptions-checkbox-label"]';
+
+export const INDEX_PATTERNS_RULE_BULK_MENU_ITEM = '[data-test-subj="indexPatternsBulkEditRule"]';
+
+export const ADD_INDEX_PATTERNS_RULE_BULK_MENU_ITEM =
+  '[data-test-subj="addIndexPatternsBulkEditRule"]';
+
+export const DELETE_INDEX_PATTERNS_RULE_BULK_MENU_ITEM =
+  '[data-test-subj="deleteIndexPatternsBulkEditRule"]';
+
+export const TAGS_RULE_BULK_MENU_ITEM = '[data-test-subj="tagsBulkEditRule"]';
+
+export const ADD_TAGS_RULE_BULK_MENU_ITEM = '[data-test-subj="addTagsBulkEditRule"]';
+
+export const DELETE_TAGS_RULE_BULK_MENU_ITEM = '[data-test-subj="deleteTagsBulkEditRule"]';
+
+export const RULES_BULK_EDIT_FORM_TITLE = '[data-test-subj="rulesBulkEditFormTitle"]';
+
+export const RULES_BULK_EDIT_FORM_CONFIRM_BTN = '[data-test-subj="rulesBulkEditFormSaveBtn"]';
+
+export const RULES_BULK_EDIT_INDEX_PATTERNS = '[data-test-subj="bulkEditRulesIndexPatterns"]';
+
+export const RULES_BULK_EDIT_OVERWRITE_INDEX_PATTERNS_CHECKBOX =
+  '[data-test-subj="bulkEditRulesOverwriteIndexPatterns"]';
+
+export const RULES_BULK_EDIT_TAGS = '[data-test-subj="bulkEditRulesTags"]';
+
+export const RULES_BULK_EDIT_OVERWRITE_TAGS_CHECKBOX =
+  '[data-test-subj="bulkEditRulesOverwriteTags"]';
+
+export const RULES_TAGS_POPOVER_BTN = '[data-test-subj="tagsDisplayPopoverButton"]';
+
+export const RULES_TAGS_POPOVER_WRAPPER = '[data-test-subj="tagsDisplayPopoverWrapper"]';
+
+export const RULES_BULK_EDIT_INDEX_PATTERNS_WARNING =
+  '[data-test-subj="bulkEditRulesIndexPatternsWarning"]';
+
+export const RULES_BULK_EDIT_TAGS_WARNING = '[data-test-subj="bulkEditRulesTagsWarning"]';
+
+export const RULES_TAGS_FILTER_BTN = '[data-test-subj="tags-filter-popover-button"]';
```

Comment:
I think this starts getting a little bit out of control :)

As a short-term measure, could you just extract all the bulk editing selectors and newly added selectors related to the rules table to, let's say, `security_solution/cypress/screens/rules/rules_table.ts`?

Thinking long-term:

- I think we should start splitting them into cohesive, meaningful modules to make Cypress tests more maintainable.
- I'd also try to group related selectors and tasks together into page objects rather than keeping them in two separate folders.

Reply:
This file already has  a lot of table utils(some actions, import, columns so on)

I leave a general table utils in `alerts_detection_rules.ts` and will move rules bulk edit methods/names out to `rules_bulk_edit.ts`